pytest_rpc/helpers.py

```diff
@@ -501,24 +502,16 @@ def parse_swift_recon(recon_out):
     """
 
     lines = recon_out.splitlines()
+    delimiter_regex = re.compile('^={79}')
     collection = []
-    data = []
-    inData = False
-    i = 0
-    while i < len(lines):
-        if not inData:
-            if lines[i].startswith('=' * 79):
-                inData = True
-        elif lines[i].startswith('=' * 79):
-            inData = False
-            collection.append(data)
-            data = []
-            # Reset counter to use the delimeter to trigger data collection in
-            # the next pass.
-            i = i - 1
-        else:
-            data.append(lines[i])
-        i = i + 1
+
+    delimiter_positions = [ind for ind, x in enumerate(lines) if re.match(delimiter_regex, x)]
```

Comment:
I'm not sure how this even works. The `delimiter_regex` is a [RegexObject](https://docs.python.org/2/library/re.html#re.RegexObject), but [re.match](https://docs.python.org/2/library/re.html#re.match) is expecting a _regex pattern_ as a string.

Instead I suggest this list comprehension:

```python
delimiter_positions = [ind for ind, x in enumerate(lines) if delimiter_regex.match(x)]
```